src-electron/util/types.js

```diff
@@ -78,6 +78,77 @@ async function typeSizeAttribute(db, zclPackageId, at, defaultValue = null) {
   }
 }
 
+/**
+ *
+ * @param {*} value
+ * @param {*} size
+ * @returns The big endian value for a given float value padded with
+ * the given size.
+ */
+function convertFloatToBigEndian(value, size) {
+  let res = ''
+  const arrayBuffer = new ArrayBuffer(size)
+  const dataView = new DataView(arrayBuffer)
+  if (size == 4) {
+    dataView.setFloat32(0, value, false)
+  } else if (size == 8) {
+    dataView.setFloat64(0, value, false)
+  } else {
+    // Throwing a warning for unknown float type and continuing with 64bit float
+    env.logWarning('Unknown Float Type value: ' + value)
+    dataView.setFloat64(0, value, false)
+  }
+  for (let i = 0; i < size; i++) {
+    let t = dataView.getUint8(i).toString(16)
+    if (t.length == 1) {
+      t = '0' + t
+    }
+    res += t
+  }
+  return '0x' + res
+}
+
+/**
+ *
+ * @param {*} value
+ * @param {*} size
+ * @returns The big endian value for a given integer value padded with
+ * the given size.
+ */
+function convertIntToBigEndian(value, size) {
+  const arrayBuffer = new ArrayBuffer(size)
+  const dataView = new DataView(arrayBuffer)
+  if (size == 1) {
+    dataView.setInt8(0, value, false)
+  } else if (size == 2) {
+    dataView.setInt16(0, value, false)
+  } else if (size == 3) {
+    dataView.setInt32(0, value, false)
+  } else if (size == 4) {
+    dataView.setInt32(0, value, false)
+  } else if (size == 5) {
+    dataView.setBigInt64(0, value, false)
```

Comment:
Are you sure this will work right?  Why would `value` be a BigInt object here?  It almost certainly won't be.....  Is the expectation that `value` is a string and hence will auto-convert to BigInt?

Reply:
I believe this should work. I was mostly thinking in terms of the range. Switching it to setInt32 for a negative integer with attribute size 5 won't work unless I use dataView.setBigInt64

Reply:
Sure.  The big question is whether this will throw an exception (which it will if `Value` is a Number: implicit conversion from Number to BigInt is not allowed in JS).  It would be safer to do `BigInt(value)`.

Reply:
Switched to BigInt